contrib/repro_test.bzl

```diff
@@ -0,0 +1,222 @@
+(being 'image') inside each of the containers. The rule then extracts the
+output files for the test image. Finally, both extracted images are compared.
+A test image is considered to be reproducible when both instances of the same
+image have the same digest and ID. On a mismatch of one of those values, the
+test fails and the container_diff tool
+(https://github.com/GoogleContainerTools/container-diff#container-diff)
+produces the summary of their differences.
+
+Args:
+    image: A container_image target relative to the project's root to test
+           for determinism.
+    base: An image target to build and reproduce the 'image 'in.
+          This image must have Bazel and docker installed and available in the PATH.
+    workspace_file: The WORKSPACE file of the project containing the 'image'
+                    target to help detect project's root path.
+    container_diff_args: (optional) Args to the container_diff tool as specified here:
+                         https://github.com/GoogleContainerTools/container-diff#quickstart
+
+Example:
+
+container_repro_test(
+    name = "set_cmd_repro_test",
+    image = "//tests/container:set_cmd",
+    base = "@bazel_latest//image",
+    workspace_file = "//:WORKSPACE",
+    container_diff_args: ["history", "file", "size", "rpm", "apt", "pip", "node"],
+)
+"""
+
+load("@base_images_docker//util:run.bzl", _extract = "extract")
+load("@bazel_skylib//lib:dicts.bzl", "dicts")
+load("//container:container.bzl", _container = "container")
+load("//skylib:filetype.bzl", container_filetype = "container")
+
+def _impl(ctx):
+    """Core implementation of container_repro_test"""
+
+    name = ctx.attr.name
+    workspace_file = ctx.file.workspace_file
+    proj_root = ctx.actions.declare_file(name + "_src_project_root.txt")
+
+    # Record the absolute path to the provided WORKSPACE file and assume it to
+    # be the root of the Bazel project to build the 'image' in.
+    ctx.actions.run_shell(
+        command = "readlink -f %s | xargs dirname >> %s" % (workspace_file.path, proj_root.path),
+        inputs = [workspace_file],
+        outputs = [proj_root],
+    )
+
+    # Tar the current source project files to mount into the image that will
+    # be used to build and repro the 'image' in.
+    proj_tar = ctx.actions.declare_file(name + "_src_project.tar")
+    ctx.actions.run_shell(
+        command = "tar -cvf %s \"$(cat %s)\"" % (proj_tar.path, proj_root.path),
+        inputs = [proj_root],
+        outputs = [proj_tar],
+    )
+
+    image_tar = ctx.actions.declare_file(name + ".tar")
+
+    # Build the Bazel image to build and repro the 'image' in.
+    # Mount the source project and its path into this image.
+    _container.image.implementation(
+        ctx,
+        files = [proj_root],
+        tars = [proj_tar],
+        output_executable = ctx.actions.declare_file(name + "_load.sh"),
+        output_tarball = image_tar,
+        workdir = "/",
```

Comment:
you can just set the workdir as the project_root (if its a var, not a file), that way you dont need to put the file in the container

Reply:
See reply [here](https://github.com/bazelbuild/rules_docker/pull/955#discussion_r301288032)

Reply:
sgtm